File: app/src/main/java/ua/hospes/rtm/domain/drivers/DriversRepository.java
package ua.hospes.rtm.domain.drivers;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;
import ua.hospes.rtm.domain.drivers.models.Driver;

/**
 * @author Andrew Khloponin
 */
public interface DriversRepository {
    Observable<Driver> get();

    Observable<Driver> get(int... ids);

    Observable<Driver> getByTeamId(int teamId);

    Observable<List<Driver>> listen();


    Observable<Boolean> save(Driver driver);

    Observable<Boolean> addDriversToTeam(int teamId, int... driverIds);

    Observable<Boolean> removeDriversFromTeam(int teamId);


    Single<Integer> remove(int id);

    Single<Integer> removeAll();
}

File: app/src/main/java/ua/hospes/rtm/domain/drivers/DriversRepository.kt
package ua.hospes.rtm.domain.drivers

import kotlinx.coroutines.flow.Flow

internal interface DriversRepository {
    suspend fun get(): List<Driver>

    suspend fun get(vararg ids: Long): List<Driver>

    suspend fun getNotInRace(teamId: Long): List<Driver>


    fun listen(): Flow<List<Driver>>


    suspend fun addDriversToTeam(teamId: Long, vararg driverIds: Long)

    suspend fun removeDriversFromTeam(teamId: Long)


    suspend fun save(driver: Driver): Driver

    suspend fun delete(id: Long)

    suspend fun clear()
}